File: src/main/java/commands/InspectorLibCommand.java
package commands;

import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public interface InspectorLibCommand {

    String AGE = "age";

    boolean parse(String[] args);

    default boolean fileOrFolderExists(String fileOrFolderPath) {
        Path path = Paths.get(fileOrFolderPath);
        boolean result = Files.exists(path) && (Files.isRegularFile(path) || Files.isDirectory(path));

        if (!result)
            System.err.println("Could not find path " + path.toAbsolutePath());

        return result;
    }

    void execute(String[] args);
}

File: src/main/kotlin/commands/InspectorLibCommand.kt
package commands

import java.io.File

const val AGE_COMMAND = "age"
const val VULNERABILITY_COMMAND = "vulnerabilities"
const val SUMMARY_COMMAND = "summary"
const val RULE_COMMAND = "rules"
const val HELP_COMMAND = "help"

abstract class InspectorLibCommand {

    abstract fun parse(args: Array<String>): Boolean

    abstract fun execute()

    abstract fun usage(): String

    fun fileOrFolderExists(fileOrFolderPath: String): Boolean {
        val file = File(fileOrFolderPath)
        if (!file.exists()) System.err.println("Could not find path $fileOrFolderPath")
        return file.exists()
    }
}